pkg/dashboard/resource/v3iostream.go

```diff
@@ -113,6 +201,141 @@ func (vsr *v3ioStreamResource) getNamespaceFromRequest(request *http.Request) st
 	return vsr.getNamespaceOrDefault(request.Header.Get("x-nuclio-project-namespace"))
 }
 
+func (vsr *v3ioStreamResource) getShardLagsMap(ctx context.Context, info *v3ioStreamInfo) (map[string]restful.Attributes, error) {
+
+	// create v3io context
+	v3ioContext, err := v3iohttp.NewContext(vsr.Logger, &v3iohttp.NewContextInput{
+
+		// allow changing http client for testing purposes
+		HTTPClient: vsr.v3ioHTTPClient,
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed creating v3io context")
+	}
+
+	// a data plane access key is required for v3io operations
+	accessKey := vsr.getCtxSession(ctx).GetPassword()
+	if accessKey == "" {
+		return nil, errors.Wrap(err, "A data plane access key is required")
+	}
+
+	dataPlaneInput := v3io.DataPlaneInput{
+		URL:           vsr.getPlatform().GetConfig().StreamMonitoring.WebapiURL,
+		AccessKey:     accessKey,
+		ContainerName: info.ContainerName,
+	}
+	getContainerContentsInput := &v3io.GetContainerContentsInput{
+		Path:             info.StreamPath,
+		GetAllAttributes: true,
+		DirectoriesOnly:  false,
+		DataPlaneInput:   dataPlaneInput,
+	}
+
+	shardLagsMap := map[string]restful.Attributes{}
+	lock := sync.Mutex{}
+
+	// we get the v3io container contents in batches until we go over everything
+	// when we find a shard content, get its lag details
+	for {
+		response, err := v3ioContext.GetContainerContentsSync(getContainerContentsInput)
+		if err != nil {
+			return nil, errors.Wrap(err, "Failed to get container subdirectories")
+		}
+
+		getContainerContentsOutput := response.Output.(*v3io.GetContainerContentsOutput)
+		response.Release()
+
+		errGroup, _ := errgroup.WithContextSemaphore(ctx,
+			vsr.Logger,
+			vsr.getPlatform().GetConfig().StreamMonitoring.GetStreamShardsConcurrentRequests)
+
+		// Iterate over subdirectories listed in the response, and look for shards
```

Comment:
The rest of the comments you added to your code start with a lowercase letter. This comment on this line is the only one starting with an uppercase letter. This is inconsistent. I find it offensive towards me, as it discriminates against people such as myself, who struggle with an undiagnosed OCD.

Reply:
My sincerest apologies to you and your kind.
I'll will treat this with the utmost importance.